addons/a11y/src/components/A11YPanel.tsx

```diff
@@ -20,60 +20,70 @@ export enum RuleType {
   INCOMPLETION,
 }
 
-const Icon = styled(Icons)(
-  {
-    height: '12px',
-    width: '12px',
-    marginRight: '4px',
-  },
-  ({ status, theme }: any) =>
-    status === 'running'
-      ? {
-          animation: `${theme.animation.rotate360} 1s linear infinite;`,
-        }
-      : {}
-);
+const RotatingIcons = styled(Icons)(({ theme }) => ({
+  height: '12px',
+  width: '12px',
+  marginRight: '4px',
+  animation: `${theme.animation.rotate360} 1s linear infinite;`,
+}));
 
-const Passes = styled.span<{}>(({ theme }) => ({
+const Passes = styled.span(({ theme }) => ({
   color: theme.color.positive,
 }));
 
-const Violations = styled.span<{}>(({ theme }) => ({
+const Violations = styled.span(({ theme }) => ({
   color: theme.color.negative,
 }));
 
-const Incomplete = styled.span<{}>(({ theme }) => ({
+const Incomplete = styled.span(({ theme }) => ({
   color: theme.color.warning,
 }));
 
-const centeredStyle = {
+const Centered = styled.span({
   display: 'flex',
   alignItems: 'center',
   justifyContent: 'center',
   height: '100%',
-};
-
-const Loader = styled(({ className }) => (
-  <div className={className}>
-    <Icon inline icon="sync" status="running" /> Please wait while the accessibility scan is running
-    ...
-  </div>
-))(centeredStyle);
-Loader.displayName = 'Loader';
-
-interface A11YPanelNormalState {
-  status: 'ready' | 'ran' | 'running';
+});
+
+interface InitialState {
+  status: 'initial';
+}
+
+interface ManualState {
+  status: 'manual';
+}
+
+interface RunningState {
+  status: 'running';
+}
+
+interface RanState {
+  status: 'ran';
+  passes: Result[];
+  violations: Result[];
+  incomplete: Result[];
+}
+
+interface ReadyState {
+  status: 'ready';
   passes: Result[];
```

Comment:
Couldn’t these 4 be made into a single interface with options for status? In other words, extending the old interface.

Reply:
It could be merged, but I made this change on purpose. I wanted to use a state machine like approach (without introducing a dedicated lib like `xstate`) to make the code more easy to reason about.

The old interface looked simple, because it was just _one_ interface, but it was (imho) harder to read and understand, because it was not very explicit. There was no differentiation between the _initial state_ and the `ready` state for example, so there was the need to mock a proper result. But inside the component you never knew, if the result was mocked or not. (Also the user briefly saw "invalid" results, because of this.)

I _could_ have added mocked results to the `initial`, `manual` and `running` states. They wouldn't "hurt", but they also wouldn't have any purpose. But it get's harder if you introduce new states which need _other_ data (like `error`).

**tl;dr**: Yes, the interface became more verbose. But it also became more correct and explicit and I hope this will make it easier to understand and extend the logic of this component.

Reply:
Long term, this addon should start using the `useAddonState` tbh.

Reply:
@ndelangen Sorry, just saw your comment now. Did I understand you correctly that the approach is fine for now and `useAddonState` should be used in a future PR? Or should I add `useAddonState` to this PR?